platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -169,16 +197,28 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
                 dispatch_async(workQueue, ^{
 #if TARGET_OS_IPHONE
                     UIImage *logoImage = [UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+                    UIImage *attributionBackgroundImage = [UIImage imageNamed:@"attribution_background" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
                     
                     UIGraphicsBeginImageContextWithOptions(mglImage.size, NO, self.options.scale);
                     
                     [mglImage drawInRect:CGRectMake(0, 0, mglImage.size.width, mglImage.size.height)];
-                    [logoImage drawInRect:CGRectMake(MGLLogoImagePosition.x, mglImage.size.height - (MGLLogoImagePosition.y + logoImage.size.height), logoImage.size.width,logoImage.size.height)];
+                    
+                    CGRect logoImageRect = CGRectMake(MGLLogoImagePosition.x, mglImage.size.height - (MGLLogoImagePosition.y + logoImage.size.height), logoImage.size.width,logoImage.size.height);
+                    [logoImage drawInRect:logoImageRect];
+                    
+                    CGPoint attributionTextPosition = CGPointMake(mglImage.size.width - 10 - attributionBackgroundSize.width,
+                                                                  logoImageRect.origin.y + (logoImageRect.size.height / 2) - (attributionBackgroundSize.height / 2));
+                    [attributionBackgroundImage drawInRect:CGRectMake(attributionTextPosition.x, attributionTextPosition.y + 1, attributionBackgroundSize.width, attributionBackgroundSize.height)
+                                                 blendMode:kCGBlendModeNormal
+                                                     alpha:0.5];
+                    [self drawAttributionText:infos origin:CGPointMake(attributionTextPosition.x + 10, attributionTextPosition.y)];
+                    
                     UIImage *compositedImage = UIGraphicsGetImageFromCurrentImageContext();
                     
                     UIGraphicsEndImageContext();
 #else
                     NSImage *logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
+                    NSImage *attributionBackgroundImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"attribution_background" ofType:@"pdf"]];
```

Comment:
attribution_background.pdf looks very simple – couldn’t we implement the same effect using image compositing APIs such as the ones provided by Core Image? Additionally, the screenshot in https://github.com/mapbox/mapbox-gl-native/pull/10352#issuecomment-341791295 shows that a simple translucent background doesn’t adequately ensure readability atop labels. Let’s try to find a way to reuse this code without necessarily using a view:

https://github.com/mapbox/mapbox-gl-native/blob/c24721103d47a3fdce6ca3cc5200d81c8f03d991/platform/macos/src/MGLMapView.mm#L387-L404